### PR TITLE
cluster-display: fetch disabled clusters on demand

### PR DESCRIPTION
**Pull request by danilo-gemoli**
Fetch disabled clusters from Prow on demand instead of getting the list at startup.
This handles the problem of a new cluster being disabled and `cluster-display` not being able to notice it unless restarted.

### CODE REVIEW

File: cmd/cluster-display/main.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"time"
 
 	"github.com/sirupsen/logrus"
+	"golang.org/x/time/rate"
```
Comment (liangxia) [inline cmd/cluster-display/main.go:17]: Looks like we also need to update `go.mod` ?

Comment (danilo-gemoli) [reply cmd/cluster-display/main.go:17]: Yes, updated. Thanks!

**Pull request merged**